VPLAY-9882:Added [CDAI] Error code and description for adResolvedEvent

srikanthreddybijjam-comcast opened the pull request:
Reason for change: Updated all the CDAI failures.
Test Procedure: Refer jira ticket VPLAY-9882
Priority: P1

Signed-off-by: srikanthreddybijjam-comcast <srikanthreddybijjam.2000@gmail.com>

p-bond committed:
VPLAY-10150 - Remove floating point comparisons in AampTsbReader

p-bond committed:
VPLAY-10150 - Remove floating point comparisons in AampTsbReader

p-bond committed:
Additional logging for playback rate

p-bond committed:
move rate init

p-bond committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-10150

p-bond committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-10150

srikanthreddybijjam-comcast committed:
VPLAY-9882:Resolved CDAI failures with ErrorCodes and ErrorDescriptions.

Reason for change:Updated all the CDAI failures.
Test Procedure: Refer jira ticket VPLAY-9882
Priority: P1

Signed-off-by: srikanthreddybijjam-comcast <srikanthreddybijjam.2000@gmail.com>

p-bond committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-10150

lashmintha committed:
VPLAY-9299:Address 200ms tune delay when using enableMediaProcessor as true

VPLAY-9299:Address 200ms tune delay when using enableMediaProcessor as true. 

Reason for change: avoid unnecessary delayed media processor flush.

Since SegmentTimeline based streams provide the PTS values in the manifest,
there is no need to parse the firstPTS from the segment and send a belated flush later.
This optimization saves nearly 280ms on XIONE-UK for DASH SegmentTimeline stream tunes

Risks: Low
Test Procedure: Refer jira ticket
Priority: P1

Signed-off-by: lashmintha <lashminthaj17@gmail.com>

pstroffolino committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-10150

p-bond committed:
Merge pull request #258 from rdkcentral/feature/VPLAY-10150

Feature/vplay 10150

pstroffolino committed:
VPLAY-10193 pssh parsing for mp4demux (#243)

VPLAY-10193 drm metadata support in mp4demux

Reason for Change: support for parsing drm-specific (pssh, senc, etc.) boxes in mp4demux, and new aampcli 'parse' command
Test Guidance: can use new 'parse' command  to demux an encrypted header/segment and see those boxes processed
Risk: Low (not used by default)

Signed-off-by: Stroffolino <philip_stroffolino@cable.comcast.com>

nu641001 committed:
VPLAY-10265: Playback Rate Correction Issue Observed in LLD Channel (#286)

VPLAY-10265: Playback Rate Correction Issue Observed in LLD Channel

Reason for change: Use to SYSTEM clock instead of STEADY clock for comparisons to publish time and manifest end for calculations of live latency, aligned to click time. avoiding bad value that can cause rate correction to wrongly be used.
Risks: Low
Test Procedure: Test with LLD channels
Priority: P1

Signed-off-by: Nandu <nu641001@gmail.com>

jfagunde committed:
VPLAY-10269 Add a script to parse L1 test output (#267)

VPLAY-10269 Add a script to parse L1 test output

Reason for Change: Identify which L1 tests failed or did not complete
Includes help and detection for skipped tests.

Test Procedure: Run the script with L1 tests console output
Risks: Low

gavin-parry committed:
VPLAY-10301 Deprecate use of GST_CHECK_VERSION(1,18,0) (#279)

Reason for change: We no longer wish to support gst < 1.18.0

Test Procedure: Ensure build fails with older gst versions. Also ensure
it builds in bitbake for a target using gst>1.18.0 and in local builds
on Ubuntu.

Risks: Low

Co-authored-by: pstroffolino <Philip_Stroffolino@cable.comcast.com>

pstroffolino committed:
VPLAY-10350 "godspell cleanup, continued"

VPLAY-10350 "godspell cleanup, continued"

Reason for Change: this fixes a small set of bogus spellings, i.e. backwordSerach, briefa, bol sttaus, senario, partiton in core aamp

Test Guidance: no godspell warnings
Risk: Low (mostly just changes in comments)

Signed-off-by: Stroffolino <philip_stroffolino@cable.comcast.com>

pstroffolino committed:
VPLAY-10193 replacement for gst_buffer_new_memdup

VPLAY-10193 replacement for gst_buffer_new_memdup

Reason for Change: avoid dependency on gstreamer 1.20
Test Guidance: no more build failure on skyxione-xoe having gst 1.18.5
RIsk: Low (already broken!)

Signed-off-by: Stroffolino <philip_stroffolino@cable.comcast.com>

pstroffolino committed:
VPLAY-10375 Revert "VPLAY-9299:Address 200ms tune delay when using enableMediaProcessor a…" (#296)

VPLAY-10375 L2 Test 8008 regression for short ads caused by VPLAY-9299

Temporary revert to avoid CDAI regression surfaced in l2 test 8008 (TEST2 and TEST3)

This reverts commit 2cb02b4ee349b19648977bab28aced90c4d79a4e.

pstroffolino committed:
VPLAY-10265: Revert "VPLAY-10265: Playback Rate Correction Issue Observed in LLD Channel (…" (#297)

VPLAY-10265: Revert "VPLAY-10265: Playback Rate Correction Issue Observed in LLD Channel (…" (#297)

This change broke several L2 tests related to trickplay.  Reverting until failure is understood and change reworked.

This reverts commit 65999c961463ba21d45329a6371b1b673ec63111.

dp0000 committed:
rdk-58554 Address the warnings in CPE integration

RDK-58554 Address the warnings in CPE integration

Reason for change: add casting to avoid compilation warning with float vs. integer play rate
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

ALSAMEEMA committed:
RDK-58568 : Address conversion mismatch in opendrmsession and pass sessionId by reference (#295)

RDK-58568 : address conversion mismatch in opendrmsession and pass sessionId by reference

Reason for change: improve integration of content protection service firebolt sdk
- addresses an issue found while working with Guru's TestCase document, where sessionId's are invalid sometimes with Firebolt flow.
- 'sessionid' was being passed by value, not reference, so was not being populated as an output parameter
- sessionid was being parsed as int, rather than 64 bit long long

Fix is specific to firebolt flow only, where the sessionId is not being properly stored back to be used later, ie) when we invoke any other api's using the session Id, those calls will fail in the downlayer,
So far no visible issue, but there is a chance of issue could happen while closing existing sessions, that might end up in sessions being created but not closed properly and also whole invoking watermark api's

Test Procedure: Refer Ticket
Risks: Medium
Priority: P1

Signed-off-by: Alsameema.A Alsameema_AhmedAnsar@Comcast.Com

Vinish100 committed:
VPLAY-9299: Address 200ms tune delay when using enableMediaProcessor as true. (#298)

Reason for change: The delayed flush in media processor adds to tune delay.
Since SegmentTimeline based streams provide the PTS values in the manifest,
there is no need to parse the firstPTS from the segment. Fixed L2 failures
in 8008
Risks: Low
Test Procedure: Refer jira ticket
Priority: P1

kd-syna committed:
VPLAY-10294 Add timestamps to aamp-cli output. (#299)

VPLAY-10249 Add timestamps to aamp-cli output.

Reason for Change:
	Allow the timing of aamp-cli log output to be accurately
	compared to the timing of aamp log output.

Test Guidance:
	Check that aamp-cli log output contains timestamps that are
	consistent with aamp's (in both the values and the formatting).

Risk: Low

anshephe committed:
VPLAY-10207: [VIPA][RIALTO] Black frames when transitioning to/from trickplay (#303)

VPLAY-10207: [VIPA][RIALTO] Black frames when transitioning to/from trickplay

Reason for Change: To avoid black frames when transitioning to/from trickplay with Rialto configuration, don't tear down pipeline, instead send eos to audio sink, to make it exit preroll.

* Had to check if rate is trickplay, as gstPrivateContext->stream[eGST_MEDIATYPE_AUDIO].format is not updated to INVALID in ConfigurePipeline
* Temporary hack to test video\audio master for SoC
* Temporary hack to test if soc IsVideoMaster is needed or just IsPlatformSegmentReady
* Temporary hack to add in incrementing number to logs, to check for missing log lines
* Removed temporary hacks
* Corrected indentation
* Put the original logic back for calculating videoOnly for property single-path-stream
* Update InterfacePlayerRDK.cpp
* refactored configureStream avoid a memset

Test Procedure: Tested with VIPA Test App

Risk: Low

---------

Signed-off-by: Andrew Shepherd anshephe@synamedia.com

Vinish100 committed:
VPLAY-10294 Add timestamps to aamp-cli output (#313)

VPLAY-10294 Add timestamps to aamp-cli output

Reason for change: fix printf format in new aampcli-specific logging macros to avoid osx compilation issus.
Test Procedure: As mentioned in the ticket
Risks: None

Signed-off-by: Vinish100 <vinish.balan@gmail.com>

rajat698 committed:
VPLAY-10344 Incorporate L1 parser script to Github actions CI

VPLAY-10344 Incorporate L1 parser script to Github actions CI

Reason for change: Adds L1 test run script to Github actions L1 tests workflow to PRs of rdkcentral/aamp repo.
 to display pretty utest run

Risks: Low
Priority: P1
---------

Signed-off-by: rajat698 <emailofrajatyadav@gmail.com>

anjali-syna committed:
VPLAY-10090 VPLAY-10399 - Audio loss in Trick to play transition in L2 tests (#288)

Reason for change: Audio is lost while performing fast forward to live
 in L2 test environment. This is because in TuneHelper,  Configure is
 happening after Flush() call with proper seek position.
 This results in pipeline loosing the flush and seek, thus affecting AV sync.

Changes:
* Perform Flush after Configure in TuneHelper
* Also, call flush with shouldTearDown false, because in case of a new tune,
  pipeline might not be in a playing/paused state which causes Flush() to
  destroy the pipeline. This has to be avoided.
* Adding new unit test:TuneHelperWithAampTsbConfigureFlushSequence

Test Procedure: Run L2 tests
Risks: Low

Signed-off-by: anjali-syna <206662904+anjali-syna@users.noreply.github.com>

pstroffolino committed:
VPLAY-10294 fix logging for 'list' command (#323)

Reason for Change: fix regression with list command formatting
Test Guidance: use 'list' command in aampcli
Risk: None

Signed-off-by: Stroffolino <philip_stroffolino@cable.comcast.com>

psiva01 committed:
VPLAY-10434:Crash (37983920-lstring::atof) during ad break of SLE (#322)

VPLAY-10434:Crash (37983920-lstring::atof) during ad break of SLE (#322) 

Reason for change: Replaced assert() with throw std::runtime_error()
exception.
Test Procedure: Refer Jira
Risks: No

Signed-off-by: psiva01 <sivasubramanian.patchaiperumal@ltts.com>

jameslofthouse-RED committed:
VPLAY-10423 Captions stop after advert (discontinuity) (#314)

VPLAY-10423 Captions stop after advert (discontinuity)  

Reason for change: Without PTS restamp enabled, text data is not sent to subtec after an ad break
Test Procedure: See ticket
Risks: Low

Signed-off-by: James Lofthouse <james_lofthouse@comcast.com>
Co-authored-by: James Lofthouse <james_lofthouse@comcast.com>

Vinish100 committed:
VPLAY-10406: Handle zero buffers with HTTP 200 in DE-LL (#324)

VPLAY-10406: Handle zero buffers with HTTP 200 in DE-LL

Reason for change: Make chunk mode injection for LLD configurable
Empty buffers with download successful is now treated as
partial file download to trigger rampdown.
Test Procedure: Check for playback freezes in DE LLD streams
Also validate with L2 1019
Risks: Low

Signed-off-by: Vinish100 <vinish.balan@gmail.com>

varshnie committed:
VPLAY-10239:L2 5003 failure (#305)

VPLAY-10239:L2 5003 failure

Reason for change:L2 5003 failure
Test Procedure: Refer jira ticket
Priority: P1

Signed-off-by: varshnie <varshniblue14@gmail.com>

jfagunde committed:
VPLAY-9399 Improve AAMP logs (#244)

VPLAY-9399 Improve AAMP logs

Reason for change: make it easier to triage various AAMP issues; more consistent use of MIL logging

Test Procedure: Verify AAMP L1 tests pass

Vinish100 committed:
VPLAY-10489: Fix L2 8006 failures due to VPLAY-9299

Reason for change: Set shouldTearDown flag to false in Flush()
call to avoid pipeline teardown and crash. Earlier the order was different
and hence the pipeline teared down by Flush would be recreated by Configure()
This is not happening anymore.
Test Procedure: Test 8006 should pass
Risks: None

Signed-off-by: Vinish100 <vinish.balan@gmail.com>

Vinish100 committed:
VPLAY-10406: Fix L2 2024 failure due to empty buffer check

Reason for change: Check for zero-byte buffer in successful downloads
Test Procedure: 2024 L2 should pass
Risks: Low

Signed-off-by: Vinish100 <vinish.balan@gmail.com>

pstroffolino committed:
VPLAY-10424 unreliable player state eventing when stopping playback

VPLAY-10424 unreliable player state event when stopping playback

Reason for Change: minimally invasive patch to address race condition with player state assignment and events while stopping
- now consistently using idle state when stopping - after flushing events.  This avoids undefined behaviors/race conditions.
- introduces an "isDestructing" parameter so that internally called stop associated with destructing a playerInstance class instance doesn't generate state change events referencing stale player instance

Test Guidance: run l2 tests (especially 1006) and get readout from apps
Risk: Medium

Signed-off-by: Stroffolino <philip_stroffolino@cable.comcast.com>

* isDestructing param

varshnie committed:
VPLAY-9810:Add droppedFrames to MonitorAV event (#270)

VPLAY-9810:Add droppedFrames to MonitorAV event

Reason for change:Added audio buffered duration as a part of mediaprogress event and dropped frames as part of monitorav event
Risks: Low
Test Procedure: Refer jira ticket
Priority: P2

Signed-off-by: varshnie varshniblue14@gmail.com

pstroffolino committed:
VPLAY-9810 monitor AV typo (#335)

VPLAY-9810 monitor AV typo

Reason for Change: fix typo caught by godspell
Test Guidance: godspell no longer flagging
Risk: None

Signed-off-by: Stroffolino <philip_stroffolino@cable.comcast.com>

Vinish100 committed:
VPLAY-10516: Fix L2 5003 failures in sprint

Reason for change: SetLocalTSBInjection before querying
GetFirstPTS to get the accurate PTS value from TSBReader
Test Procedure: All L2 5003 should pass
Risks: Low

Signed-off-by: Vinish100 <vinish.balan@gmail.com>

supriyadappu committed:
VPLAY-10523 : [AAMP] Sprint release version is displayed as 7.06 instead of 7.07 (#336)

VPLAY-10523 : [AAMP] Sprint release version is displayed as 7.06 instead of 7.07 (#336)

Reason for change: Changed AAMP version to 7.07
Test Procedure: Refer Jira
Risks: None

nebutler963 committed:
VPLAY-9840 -AAMP TSB - OOB subtitles – L1 Tests (#237)

VPLAY-9840 -AAMP TSB - OOB subtitles – L1 Tests

Reason for Change: L1 Test for OOB subtitles support when using TSB.
Test Procedure: L1
Risks: Low
---------

Co-authored-by: pstroffolino <Philip_Stroffolino@cable.comcast.com>

jfagunde committed:
VPLAY-10510 Improve L1 test results parsing script (#337)

VPLAY-10510 Improve L1 test results parsing script

Reason for change: Fix wrong results shown in GitHub actions
Summary of changes:
* Update regex patterns to handle complex test names
* Detect timeout failures
* Improve disabled test detection
* Use sets to better performance
* Show specific failure type ([Failed], [Timeout]) in the output
* Fix completed tests incorrectly flagged as incomplete

Test Procedure: Run script with the output of running L1 tests
Risks: Low

Co-authored-by: pstroffolino <Philip_Stroffolino@cable.comcast.com>

pstroffolino committed:
VPLAY-10512 coverity fixes (#333)

VPLAY-10512 coverity fixes

Reason for Change: collection of coverity flagged patches

Test Guidance: general regression

Risk: Low

Signed-off-by: Stroffolino <philip_stroffolino@cable.comcast.com>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/rdkcentral/aamp?pullRequest=353) <br/>All committers have signed the CLA.

srikanthreddybijjam-comcast committed:
VPLAY-9882:Resolved CDAI failures with ErrorCodes and ErrorDescriptions.

Reason for change:Updated all the CDAI failures.
Test Procedure: Refer jira ticket VPLAY-9882
Priority: P1

Signed-off-by: srikanthreddybijjam-comcast <srikanthreddybijjam.2000@gmail.c
om>

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
f6eeb33ad2f6f3c3baf50a4663f07788
PRESENT
-->

# Coverity Issue - Data race condition
A wait is performed without a loop. If there is a spurious wakeup, the condition may not be satisfied.

_Medium Impact, CWE-none_
_BAD_CHECK_OF_WAIT_COND_

## How to fix
 Check the wait condition in a loop, with the lock held. The lock must not be released between the condition and the wait.

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:7479](undefined//blob/undefined/priv_aamp.cpp#L7479)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
8c7810d0c1174017f276b9c15a67eb68
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"effectiveUrl" is copied in call to copy assignment for class "std::string", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""effectiveUrl"")" instead of "effectiveUrl".

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[admanager_mpd.cpp:939](undefined//blob/undefined/admanager_mpd.cpp#L939)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
92516652040f0ab084cf0787b7b242bf
PRESENT
-->

# Coverity Issue - Unchecked return value
Calling "GetFile" without checking return value (as is done elsewhere 8 out of 9 times).

_Medium Impact, CWE-252_
_CHECKED_RETURN_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[admanager_mpd.cpp:936](undefined//blob/undefined/admanager_mpd.cpp#L936)

rdkcmf-jenkins commented on streamabstraction.cpp:
```diff
@@ -3389,12 +3395,16 @@ void MediaTrack::OnSinkBufferFull()
+		}
+		
 		// check if cache buffer is full and caching was needed
 		if (IsFragmentCacheFull() && (eTRACK_VIDEO == type) &&
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
fa66964f0e9f006f23d564ed80411b71
PRESENT
-->

# Coverity Issue - Data race condition
Accessing "this->cachingCompleted" without holding lock "MediaTrack.mutex". Elsewhere, "MediaTrack.cachingCompleted" is written to with "MediaTrack.mutex" held 3 out of 3 times.

_Medium Impact, CWE-366_
_MISSING_LOCK_

rdkcmf-jenkins commented on AampEvent.cpp:
```diff
@@ -150,14 +150,16 @@
-		mBufferedDuration(bufferedDuration),
+		mVideoBufferedDurationMs(videoBufferedDuration),
+		mAudioBufferedDurationMs(audioBufferedDuration),
 		mSEITimecode(seiTimecode),
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
b847c24d2d0738430234b752556c59dd
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"seiTimecode" is passed-by-value as parameter to "std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const &)", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""seiTimecode"")" instead of "seiTimecode".

rdkcmf-jenkins commented on test/gstTestHarness/gst-test.cpp:
```diff
@@ -175,56 +172,59 @@ void GetVideoSegmentPath( char path[MAX_PATH_SIZE], int segmentNumber, VideoReso
+					{
+						gMp4Demux[mediaType] = new Mp4Demux();
+					}
+					gMp4Demux[mediaType]->Parse(ptr,len);
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
6544e5df56acd0c74e03653967055800
PRESENT
-->

# Coverity Issue - Untrusted loop bound
Passing tainted expression "*this->ptr" to "Parse", which uses it as a loop boundary.

_Medium Impact, CWE-606_
_TAINTED_SCALAR_

## How to fix
 Ensure that tainted values are properly sanitized, by checking that their values are within a permissible range.
